### PR TITLE
Push the new STI images into configured output imageRepository

### PR DESCRIPTION
**Pull request by mfojtik**
@bparees this will push the image to defined imageRepository after successful build.

### CODE REVIEW

File: images/builder/docker/sti-image-builder/bin/build.sh [hunk 1]
```diff
@@ -47,16 +47,15 @@ pushd $source_dir >/dev/null
...
-
-# Tag the image with the GIT ref if the SOURCE_REF is set
-#
-if ! [ -z "${SOURCE_REF}" ]; then
-  docker tag ${image_id} ${IMAGE_NAME}:git-$SOURCE_REF
+if ! [ -z "${OUTPUT_REGISTRY}" ]; then
+  image_id=$(docker inspect --format="{{ .Id }}" ${IMAGE_NAME}-candidate:latest)
+  echo "Pushing the ${OUTPUT_REGISTRY}/${OUTPUT_IMAGE} image..."
+  OUTPUT_IMAGE=${OUTPUT_IMAGE:-$IMAGE_NAME}
+  set -x
```
Comment (bparees) [inline images/builder/docker/sti-image-builder/bin/build.sh]: is this intentional?

Comment (mfojtik) [inline images/builder/docker/sti-image-builder/bin/build.sh]: i used that for debugging and I thought it might be useful to show what is docker doing

**Comment by bparees** [comment]
lgtm.

**Comment by mfojtik** [comment]
[merge]

File: images/builder/docker/sti-image-builder/bin/build.sh [hunk 2]
```diff
@@ -47,16 +47,15 @@ pushd $source_dir >/dev/null
...
-docker tag ${image_id} ${IMAGE_NAME}:qa-ready
-
-# Tag the image with the GIT ref if the SOURCE_REF is set
-#
-if ! [ -z "${SOURCE_REF}" ]; then
-  docker tag ${image_id} ${IMAGE_NAME}:git-$SOURCE_REF
+if ! [ -z "${OUTPUT_REGISTRY}" ]; then
+  image_id=$(docker inspect --format="{{ .Id }}" ${IMAGE_NAME}-candidate:latest)
+  echo "Pushing the ${OUTPUT_REGISTRY}/${OUTPUT_IMAGE} image..."
+  OUTPUT_IMAGE=${OUTPUT_IMAGE:-$IMAGE_NAME}
```
Comment (smarterclayton) [inline images/builder/docker/sti-image-builder/bin/build.sh]: Al of the arguments in this file need quotes around them, or you are risking an injection attack.

Comment (mfojtik) [inline images/builder/docker/sti-image-builder/bin/build.sh]: good catch, this was also the reason why pushing a 'qe-ready' tag failed

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**